### ISSUE TITLE
Allow the same node to be matched multiple times with different tags

### ISSUE THREAD

**Issue by JustusFluegel** [Nadrieril/pest_consume#16]
Allow the following:

```rs
match_nodes!{
    [tag#node_1(a),tag_1#node_2(b)..,tag_2#node_2(c)..] => todo!();
};
```

with for example the grammar

```pest
input = { #tag = node_1 ~ ";" ~ #tag_1 = node_2*~ ";" ~ #tag_2 = node_2* }
```

**Comment by Nadrieril** [comment]
What are these tag things? Is that a new feature in pest?

**Comment by JustusFluegel** [comment]
yes it is http://pest.rs/book/grammars/syntax.html#tags

**Comment by Nadrieril** [comment]
I see. Did you mean to add repetition to `tag_2` in your example, e.g. `[tag#node_1(a), tag_1#node_2(b).., tag_2#node_2(c)..]`? Do I correctly understand that this allows for multiple repetition patterns?

**Comment by Nadrieril** [comment]
Hm I guess we already could allow multiple repetition patterns today, but this allows it for the same node. That's interesting

**Comment by JustusFluegel** [comment]
yea that's basically it

**Comment by JustusFluegel** [comment]
Same as with #15 if you don't feel like dealing with this but think this is a welcome addition in general feel free to let me know.

**Comment by Nadrieril** [comment]
Thanks for offering! I'll give it a go because it seems easy enough, I'll ping you if I'd rather you do it instead :)

**Comment by JustusFluegel** [comment]
sure that works :+1:

**Comment by JustusFluegel** [comment]
probably multiple repetition patterns need to be unique over the tuple `(Node,Option<Tag>)` I guess

**Comment by Nadrieril** [comment]
Yeah, I'm thinking I'll just do it greedily, so if someone does `[node1(a).., node1(b)..]`, `b` will be empty

**Comment by JustusFluegel** [comment]
or that probably also works fine :+1:

**Comment by Nadrieril** [comment]
Alright, I've got tag support on `master`! Not yet multiple `..` patterns

**Comment by Nadrieril** [comment]
Alright, multi multi patterns are on master! Can you test that this does what you want?

**Comment by JustusFluegel** [comment]
Sure I'll do so sometime next week (hopefully tomorrow) since I am kinda busy right now.
